keyboards/trashman/ketch/keymaps/jetpacktuxedo/keymap.c

```diff
@@ -0,0 +1,85 @@
+    ),
+    [2] = LAYOUT_jetvan( /* LAYER 2 */
+        KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
+        KC_TRNS, KC_UNDS, KC_PLUS, KC_COLN, KC_DQUO, KC_TRNS, KC_TRNS, KC_TRNS, KC_UNDS, KC_PLUS, KC_DQUO, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_LCBR, KC_RCBR, KC_PGUP, KC_PIPE,
+        KC_TRNS, TG(4),   MO(3),                              KC_TRNS,                   KC_HOME, KC_PGDN, KC_END
+    ),
+    [3] = LAYOUT_jetvan( /* LAYER 3 */
+        KC_TRNS, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_F11,  KC_F12,  KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS,   KC_TRNS,                          KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [4] = LAYOUT_jetvan( /* Gaming Layer*/
+        KC_ESC,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_WH_U, KC_BTN1, KC_MS_U, KC_BTN2, KC_TRNS, KC_TRNS,
+        KC_TAB,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_WH_D, KC_MS_L, KC_MS_D, KC_MS_R, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, LT(5, KC_SLSH),
+        KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [5] = LAYOUT_jetvan( /* RESET Layer*/
+        RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+void keyboard_post_init_user(void) {
+    #ifdef RGBLIGHT_ENABLE
+        // Set up RGB effects on the rightmost 8 leds (which are the first 8)
+        rgblight_set_effect_range(0, 8);
+        // Litearlly just doing this to lower the brightness. Since I am about to
+        // turn on the rainbow effect which changes the hue I don't actually care
+        // what that gets set to, but I am also lowering the saturation to get more
+        // of a pastel feel
+        rgblight_sethsv_noeeprom(0, 192, 128);
+        // Set LED effects to rainbow
+        rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_SWIRL);
+
+        // Init the leftmost two (last two) LEDs to a static color to use for indicators
+        setrgb(0, 0, 0, (LED_TYPE *)&led[8]);
+        setrgb(0, 0, 0, (LED_TYPE *)&led[9]);
+        rgblight_set();
+    #endif //RGBLIGHT_ENABLE
```

Comment:
This stuff could be replaced with the rgblight layers feature.

Reply:
Huh, that feature is totally new to me. I like that it also allows the layer color effects to run on top of whatever normal effects are running, too, so it solves the issue of having to exclude those from the effect section of the chain. I can't tell from the docs though if it enables color blending for layers? I specifically implemented this to use individual color channels in order to allow layers to be "stacked" or blended visually on a single LED. So on layer 1 the first LED is red and on layer 2 it is green, but if both layer 1 and layer 2 are active you get sort of a yellow as it blends both layer colors. It's not a huge deal (and I'll probably flip to rgblight layers anyway because they can sit on top of the effects and simplify all of this), just wondering if that feature allows for this at all.

Reply:
> Since the different lighting layers overlap, the order matters in the array, with later layers taking precedence

It _sounds_ like no, they won't do color blending :/

Still gonna play around with it a bit and see if I can get something I like.

Reply:
Alright, so I think this is actually _more_ code than I had before, but it is more standard. I think allowing the effect to run underneath the indicators is probably worth losing the color blending, but being able to blend the colors instead of stacking them would be a neat thing to consider for the future of the `RGBLIGHT_LAYERS` feature 😉